realm/realm-library/src/androidTest/java/io/realm/RealmListTests.java

```diff
@@ -697,261 +634,133 @@ public void where_afterContainerObjectRemoved() {
 
-        checkMethodsOnEmptyList(testRealm, owner.getDogs());
+        assertNotNull(firstDog);
     }
 
     @Test
     public void clear() {
-        Owner owner = testRealm.where(Owner.class).findFirst();
-        testRealm.beginTransaction();
-        assertEquals(TEST_OBJECTS, owner.getDogs().size());
+        Owner owner = realm.where(Owner.class).findFirst();
+        realm.beginTransaction();
+        assertEquals(TEST_SIZE, owner.getDogs().size());
         owner.getDogs().clear();
         assertEquals(0, owner.getDogs().size());
-        testRealm.commitTransaction();
+        realm.commitTransaction();
     }
 
     @Test
-    public void clear_NotDeleting() {
-        Owner owner = testRealm.where(Owner.class).findFirst();
-        testRealm.beginTransaction();
-        assertEquals(TEST_OBJECTS, testRealm.allObjects(Dog.class).size());
+    public void clear_notDeleting() {
+        Owner owner = realm.where(Owner.class).findFirst();
+        realm.beginTransaction();
+        assertEquals(TEST_SIZE, realm.allObjects(Dog.class).size());
         owner.getDogs().clear();
-        assertEquals(TEST_OBJECTS, testRealm.allObjects(Dog.class).size());
-        testRealm.commitTransaction();
-    }
-
-    @Test
-    public void contains() {
-        Owner owner = testRealm.where(Owner.class).findFirst();
-        Dog dog = owner.getDogs().get(0);
-        assertTrue("Should contain a particular dog.", owner.getDogs().contains(dog));
+        assertEquals(TEST_SIZE, realm.allObjects(Dog.class).size());
+        realm.commitTransaction();
     }
 
-    /**
-     * Test to see if a particular item that does exist in the same Realm does not
-     * exist in a query that excludes said item.
-     */
     @Test
-    public void contains_sameRealmNotContained() {
-        RealmResults<Dog> dogs = testRealm.where(Dog.class)
-                .equalTo("name", "Dog 1").or().equalTo("name", "Dog 2").findAll();
-        Dog thirdDog = testRealm.where(Dog.class)
-                .equalTo("name", "Dog 3").findFirst();
-        assertFalse("Should not contain a particular dog.", dogs.contains(thirdDog));
-    }
-
-    @Test
-    public void contains_inNonManagedList() {
-        Owner owner = testRealm.where(Owner.class).findFirst();
-        RealmList<Dog> managedDogs = owner.getDogs();
-        // Create a non-managed RealmList
-        RealmList<Dog> nonManagedDogs
-                = new RealmList<Dog>(managedDogs.toArray(new Dog[managedDogs.size()]));
-        Dog dog = managedDogs.get(0);
-        assertTrue("Should contain a particular dog", nonManagedDogs.contains(dog));
-    }
+    public void setList_clearsOldItems() {
+        realm.beginTransaction();
+        CyclicType one = realm.copyToRealm(new CyclicType());
+        CyclicType two = realm.copyToRealm(new CyclicType());
+        two.setObjects(new RealmList<CyclicType>(one));
+        two.setObjects(new RealmList<CyclicType>(one));
```

Comment:
This test is not good enough. `two.setObjects(new RealmList<CyclicType>(one)); ` twice could work even when there are bugs. maybe set a size=1 list first then size=2 list